rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -1012,68 +1018,36 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 		SubscribeTooltip(ui->gb_wakeupDelay, tooltips.settings.wake_up_delay);
 	}
 
-	// lib options tool tips
-	SubscribeTooltip(ui->lib_manu, tooltips.settings.libraries_manual);
-	SubscribeTooltip(ui->lib_both, tooltips.settings.libraries_both);
-	SubscribeTooltip(ui->lib_lv2,  tooltips.settings.libraries_liblv2);
-	SubscribeTooltip(ui->lib_lv2b, tooltips.settings.libraries_liblv2both);
-	SubscribeTooltip(ui->lib_lv2l, tooltips.settings.libraries_liblv2list);
-
-	// creating this in ui file keeps scrambling the order...
-	QButtonGroup *lib_mode_bg = new QButtonGroup(this);
-	lib_mode_bg->addButton(ui->lib_manu, static_cast<int>(lib_loading_type::manual));
-	lib_mode_bg->addButton(ui->lib_both, static_cast<int>(lib_loading_type::hybrid));
-	lib_mode_bg->addButton(ui->lib_lv2,  static_cast<int>(lib_loading_type::liblv2only));
-	lib_mode_bg->addButton(ui->lib_lv2b, static_cast<int>(lib_loading_type::liblv2both));
-	lib_mode_bg->addButton(ui->lib_lv2l, static_cast<int>(lib_loading_type::liblv2list));
-
-	m_emu_settings->EnhanceRadioButton(lib_mode_bg, emu_settings_type::LibLoadOptions);
-
-	std::vector<std::string> loadedLibs = m_emu_settings->GetLoadedLibraries();
+	std::vector<std::string> loadedLibs = m_emu_settings->GetLibrariesControl();
 
 	std::set<std::string_view> set(loadedLibs.begin(), loadedLibs.end());
 
-	for (const auto& lib : set)
-	{
-		QListWidgetItem* item = new QListWidgetItem(qsv(lib), ui->lleList);
-		item->setFlags(item->flags() | Qt::ItemIsUserCheckable); // set checkable flag
-		item->setCheckState(Qt::Checked); // AND initialize check state
-		ui->lleList->addItem(item);
-	}
-
-	extern const std::unordered_map<std::string_view, int> g_prx_list;
-
 	for (const auto& lib : g_prx_list)
 	{
-		if (set.count(lib.first))
-		{
-			continue;
-		}
+		// -1: Override LLE
+		// 1: Override HLE
+		// 0: No override
+		const int res = static_cast<int>(set.count(std::string(lib.first) + ":hle") - set.count(std::string(lib.first) + ":lle"));
 
 		QListWidgetItem* item = new QListWidgetItem(qsv(lib.first), ui->lleList);
 		item->setFlags(item->flags() | Qt::ItemIsUserCheckable); // set checkable flag
-		item->setCheckState(Qt::Unchecked); // AND initialize check state
+		item->setCheckState(res && res != (lib.second * 2 - 1) ? Qt::Checked : Qt::Unchecked); // AND initialize check state
```

Comment:
it's not really clear to the reader what you're doing here

Reply:
Added comment.